app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        key = args
+
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[key]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[key] = result
+            return result
```

Comment:
return `cache_storage[key]` once at the end of the function to avoid calling return twice